CS: convert CRLF to LF

piotr-cz opened the pull request:
### Summary of Changes

Changed EOL from CRLF (Windows) to LF (Unix/ OSX) format

Didn't change for:
- `libraries/phputf8/README` (external library)
- `media/editors/codemirror/lib/addons.css` (false positive)
- `media/editors/codemirror/lib/addons.js` (false positive)

Script I've used:

``` sh
find . -not -type d -exec file "{}" ";" | grep CRLF
```

credit: http://stackoverflow.com/a/73969/1012616
### Testing Instructions

none
### Documentation Changes Required

none

piotr-cz committed:
convert CRLF to LF

zero-24 commented:
@piotr-cz can you tak a look inti the merge conflicts?

piotr-cz commented:
resolved, but by accident I've merged master branch instead of staging

brianteeman commented:
Shouldnt the language files be fixed at the source and not here. Otherwise the changes will probably be undone when they are next updated

piotr-cz commented:
Yes, should be fixed at the source.
I wasn't sure if installation language files are updated by to Joomla or language contributors.
How can I contact contributors to `ckb-IQ` installation language?

There are links to translation teams on [community page](https://community.joomla.org/translations/joomla-3-translations.html).

I've tried to contact contributors to `pl-PL` but there are links to forum profile and no email, and I haven't received a reply yet.

infograf768 commented:
@piotr-cz 
It is OK to solve these here and merge.
The TT Coordination will remind TTs to make sure they save their ini files with LF.

photodude commented on tests/unit/suites/libraries/joomla/mediawiki/JMediawikiHttpTest.php:
```diff
@@ -1,121 +1,121 @@
+	protected function setUp()
+	{
+		$this->options = new JRegistry;
+		$this->transport = $this->getMock('JHttpTransportStream', array('request'), array($this->options), 'CustomTransport', false);
```
Please use the `$this->getMockBuilder()` rather than `$this->getMock()` directly. See #12990 for examples

photodude commented on tests/unit/suites/libraries/joomla/mediawiki/JMediawikiImagesTest.php:
```diff
@@ -1,169 +1,169 @@
+	protected function setUp()
+	{
+		$this->options = new JRegistry;
+		$this->client = $this->getMock('JMediawikiHttp', array('get', 'post', 'delete', 'patch', 'put'));
```
Please use the `$this->getMockBuilder()` rather than `$this->getMock()` directly. See #12990 for examples

photodude commented on tests/unit/suites/libraries/joomla/mediawiki/JMediawikiObjectTest.php:
```diff
@@ -1,97 +1,97 @@
+	protected function setUp()
+	{
+		$this->options = new JRegistry;
+		$this->client = $this->getMock('JMediawikiHttp', array('get', 'post', 'delete', 'patch', 'put'));
```
Please use the `$this->getMockBuilder()` rather than `$this->getMock()` directly. See #12990 for examples

photodude commented on tests/unit/suites/libraries/joomla/mediawiki/JMediawikiPagesTest.php:
```diff
@@ -1,160 +1,160 @@
+	protected function setUp()
+	{
+		$this->options = new JRegistry;
+		$this->client = $this->getMock('JMediawikiHttp', array('get', 'post', 'delete', 'patch', 'put'));
```
Please use the `$this->getMockBuilder()` rather than `$this->getMock()` directly. See #12990 for examples

photodude commented on tests/unit/suites/libraries/joomla/mediawiki/JMediawikiSearchTest.php:
```diff
@@ -1,125 +1,125 @@
+	protected function setUp()
+	{
+		$this->options = new JRegistry;
+		$this->client = $this->getMock('JMediawikiHttp', array('get', 'post', 'delete', 'patch', 'put'));
```
Please use the `$this->getMockBuilder()` rather than `$this->getMock()` directly. See #12990 for examples

photodude commented on tests/unit/suites/libraries/joomla/mediawiki/JMediawikiTest.php:
```diff
@@ -1,189 +1,189 @@
+	protected function setUp()
+	{
+		$this->options = new JRegistry;
+		$this->client = $this->getMock('JMediawikiHttp', array('get', 'post', 'delete', 'patch', 'put'));
```
Please use the `$this->getMockBuilder()` rather than `$this->getMock()` directly. See #12990 for examples

photodude commented on tests/unit/suites/libraries/joomla/mediawiki/JMediawikiUsersTest.php:
```diff
@@ -1,139 +1,139 @@
+	protected function setUp()
+	{
+		$this->options = new JRegistry;
+		$this->client = $this->getMock('JMediawikiHttp', array('get', 'post', 'delete', 'patch', 'put'));
```
Please use the `$this->getMockBuilder()` rather than `$this->getMock()` directly. See #12990 for examples

photodude commented:
You probably can skip making the changes to use `$this->getMockBuilder()` rather than `$this->getMock()` directly. and just fix the merge conflicts after #12990 is merged into staging.

Commit:
Unit test fixes

brianteeman commented:
@piotr-cz can you look at the merge conflicts please

infograf768 committed:
Updating install.xml for new fields language files (#13111)

brianteeman committed:
Update strings to be consistent with their use elsewhere in Joomla

brianteeman committed:
Make desc into a label

Commit:
Implement Countable interface in JFeed

alikon committed:
[com_content] - admin fix SQL error when apply empty filter

#13114

alikon committed:
models articles 

models articles fix

alikon committed:
model featured

model featured

alikon committed:
cs

cs

alikon committed:
ops

ops

laoneo committed:
Change type of default value back to textarea (#12811)

laoneo committed:
[com_fields] Remove obsolete custom fields code (#12803)

Commit:
Adding first batch of locales for new calendar

brianteeman committed:
[com_fields] lib.joomla.ini sync (#13116)

* Change option to value and capitalise on the label for consistency

* sync

* missing period

* typo

Commit:
Check that extension that owns a category is enabled

Commit:
More calendar language files.

Commit:
changing en-US color to default en-GB colour

Easy fix on review
changing en-US `color` to default en-GB `colour`

wilsonge committed:
Merge pull request #13129 from MATsxm/patch-2

changing en-US color to default en-GB colour

Commit:
Fixing installation SQL for PostgreSQL

jeckodevelopment committed:
Merge pull request #13136 from Bakual/FixPostgreSQL

Fixing installation SQL for PostgreSQL

wilsonge committed:
Merge pull request #13125 from chrisdavenport/ss-fix-13118

Smart Search Indexes Categories for Disabled Components

wilsonge committed:
Merge pull request #13121 from chrisdavenport/rss-countable

Implement Countable interface in JFeed

wilsonge committed:
Merge pull request #13100 from ciar4n/font-weight

Isis font-weight 200 override

wilsonge committed:
Merge pull request #12428 from photodude/patch-9

crypt-blowfish salt is too short

wilsonge committed:
Merge pull request #13123 from alikon/patch-82

[com_content] - fix empty filter sql error

wilsonge committed:
Merge pull request #13119 from brianteeman/feed

Make option desc into a label

wilsonge committed:
Merge pull request #12611 from brianteeman/TOS

TOS incorrect to check for an article

wilsonge committed:
Merge pull request #13117 from brianteeman/alt-text

[com_fields][com_baners] Alt text label and desription

wilsonge committed:
Merge pull request #12490 from brianteeman/smart_search

Allow empty search - uses wrong value

wilsonge committed:
Merge pull request #12471 from brianteeman/update_emails

Check to ensure user is not disabled before sending the email update notification

wilsonge committed:
Merge pull request #12385 from mbabker/update-restore

Updated restore.php for updates

matrikular committed:
Fix placeholder showing "Select a User" when in readonly mode (e.g. modified_by) (#12627)

* Fix placeholder showing "Select a User" when in readonly mode (e.g. modified_by)

Additional changes:
- Remove obsolete extracted variables that are not present for this field type
- Correct variable name in doc block to match extracted name
- Refactor code to remove ternary overuse

* Changed getExcluded method return value to match doc block

Additional changes:
- Satisfy Code Sniffer (e.g. @since)
- Remove else block after variable initialisation

* Change htmlspecialchars($VARNAME, ENT_COMPAT, 'UTF-8') to $this->escape($VARNAME)

Additional changes:
- Change JHtml::script() call to JHtml::_('script' ...) to allow overrides via custom register method

*  Fix placeholder showing "Select a User" when in readonly mode (e.g. modified_by)

Additional changes:
- Change htmlspecialchars($VARNAME, ENT_COMPAT, 'UTF-8') to $this->escape($VARNAME)
- Refactor code to remove the ternary overuse
- Change JHtml::script call to JHtml::_('script' ...) to allow overrides via custom register method
- Correct variable name ($exclude) in doc block to match extracted name
- Remove obsolete extracted variables that are not present for this field type
- Removed the class suffix from the hidden field
- Remove (Fix) obsolete whitespace in class attribute for the visible field when no suffix was entered

* Move hidden field outside the condition for now

* Re-Added interface

Missed that in the manual merge process. While I was working on the PR, the custom fields code got merged. Thanks @wilsonge.

* Add custom class back to hidden field for potential b/c breaks

Commit:
Support for sqlsrv

Commit:
Mering calendar locales

brianteeman committed:
resolve conflict

brianteeman committed:
resolve conflict (reverted from commit 31e131f674d4fd2857d4b840456aa586616dbb62)

brianteeman committed:
resolve conflict

wilsonge committed:
Merge branch 'staging' into show_tag_num_items

Commit:
[com_fields] No need for an alias in fields groups. (#13115)

* No need for an alias in fields groups.

* Remove alias from SQL structure.

alikon committed:
[com_menu] - delete reference on #_modules_menu deleting a menu item (#13126)

Commit:
Add a static flag to check if the event is registered (Fix #13124) (#13147)

marcochirienti committed:
Coding style - com_newsfeeds (#13139)

* Leave one white-space before the PHP code closing tag.

* Fixed wrong place for space

alikon committed:
[plg_user] - delete messages from deleted user (#13052)

see #12883

alikon committed:
Merge pull request #25 from csthomas/alikon-patch-77

Support for sqlsrv

Commit:
three more calendar languages

wilsonge committed:
Ensure we have a instance of JApplicationCms (#8166)

* Ensure we have a instance of JApplicationCms

This allows the creation of users via CLI scripts that currently error because the application object is an instance of JApplicationCli

* Change to method exists check

ciar4n committed:
[3.7.x] Input field invalid highlight (#12443)

* Session field invalid interger

* Input field invalid highlight

* Revert to original invalid highlighting

alikon committed:
[com_content] - deleted featured articles are not deleted from  #__content_frontpage table (#12505)

* [com_content] - deleted featured articles are not deleted from #__content_frontpage table

deleted featured articles are not deleted from #__content_frontpage table

* tabs

tabs

* tabs again

tabs again

* fully delete featured

delete featured from #__content_frontpage

* delete featured

delete featured

* moved to controller/models

moved to controller/models

* deploy version

deploy version

* 1 more space

1 more space

* travis CS

travis CS

* removed postDeleteHook() to solve conflicts

removed postDeleteHook() to solve conflicts and to make it better

* remove old pr way

remove old pr way

* implementation with a method override as suggested

implementation with a method override as suggested

Devportobello committed:
[NF] UX/UI Module menu assignment item type (#12752)

* Module menu assignment item type

UX UI improvment in module menu assignment to see menu item of type
separator, heading, alias

* add url type

* disabled attribute

alikon committed:
[categories] - respect  user level access for categories (#12931)

* [categories] - respcet  user level access

respcet category user level access

* [categories] - respect user access level

[categories] - respect user access level

* categories - respect user access level

respect user access level

* [categories] - respect categories access level when create a new article

respect categories access level when create a new article

* respect categories access level

search filter in article manager

andrepereiradasilva committed:
[com_installer] review config global default options (#12964)

* defaults

* sincronize with staging

* fix conflicts

Commit:
Regression: Bring back translateformat feature to JFormFieldCalendar (#12969)

* Regression: Bring back translateformat feature to JFormFieldCalendar

* Restore the missing part

* Two strings were missing in frontend.

andrepereiradasilva committed:
[com_menus] review config global default options (#12971)

* config.xml

* Update joomla.sql

* Update joomla.sql

* Update joomla.sql

* sincronize with staging

* fix conflicts

okonomiyaki3000 committed:
Move the checkboxes to the leftmost column in detail view (#12997)

alikon committed:
[com_menus] - wrong group by when multilanguages enabled - postgresql (#12999)

Pull Request for fix regression from #12051 .

Summary of Changes

fixed wrong group by when multilanguages enabled

Testing Instructions

latest staging fresh multilanguage install

Go to Menu -> All menu items

infograf768 committed:
Displaying the correct tip in Global Configuration Permissions (#13143)

* Displaying the correct tip in Global Configuration Permissions

* simplification

Fedik committed:
Disable Chosen.js for the rule field, again (#12877)

* Revert the check for instance type of data-chosen, to allow to disable Chosen for field with data-chosen="true"

* Add some explanation

frankmayer committed:
Replace deprecated isSite() and isAdmin() calls

frankmayer committed:
Fixed accidental replacements in tests

frankmayer committed:
Rev reverted replacements in external projects' files

frankmayer committed:
Corrected an oversight

frankmayer committed:
Resolve?

frankmayer committed:
Try to fix pagination-test failure

frankmayer committed:
Try to fix pagination-test failure, take2

frankmayer committed:
Try to fix pagination-test failure,  final (hopefully)

frankmayer committed:
Some more replacements

infograf768 committed:
Multilanguage: correcting saving menu item with same alias when language is changed to ALL (#13112)

* Multilanguage: correcting saving menu item with same alias when language
is changed to ALL

* not syncing anymore

* new string is no more necessary

* thanks @andre

marcochirienti committed:
Coding style - PHP code tag and control structures (#13141)

* Leave one white-space after the PHP code opening tag.

* Leave one white-space before the PHP code closing tag.

* Fixed style for control structures alternative syntax.

* Removed unnecessary space before PHP closing tag

* Removed unnecessary space before php closing tag

* Added missing space before PHP code closing tag

infograf768 committed:
Multilingual: correcting alternates when associated menu items are com_users items (#13152)

* Multilingual: correcting alternates when associated menu items are
com_users items

* simplification

* Update languagefilter.php

oops

photodude committed:
[Unit Test improvements] Use getMockBuilder rather than getMock directly  (#12990)

* Use getMockBuilder rather than getMock directly

Avoid situations that would cause `PHP Fatal error:  Call to protected method PHPUnit_Framework_TestCase::getMock()` on newer versions of phpunit

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use the correct $this object

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Set up a more mock JTableExtension with getMockBuilder

* Use getMockBuilder rather than getMock directly

* Use the correct $this object

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* remove the method after replacing

* Use getMockBuilder rather than getMock directly

* More getMockBuilder in place of getMock

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Fix testParse mock Constructor Args

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* disableOriginalConstructor

* disableOriginalConstructor()

* disableOriginalConstructor()

* disableOriginalConstructor()

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* another spot for getMockBuilder

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* remove duplicate $this->transport

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Use getMockBuilder rather than getMock directly

* Add 'setUserState' to MockApplicationCms

Fixes 2 warnings on phpUnit 5.6:
1) JModelListTest::testGetuserstateUsesDefault
Trying to configure method "setUserState" which cannot be configured because it does not exist, has not been specified, is final, or is static
2) JModelListTest::testGetuserstateUsesRequestData
Trying to configure method "setUserState" which cannot be configured because it does not exist, has not been specified, is final, or is static

* mockDatabase fails for mockTableExtension on hhvm

Fixes 10 hhvm failures that have one of the following messages
```php
UnexpectedValueException: No columns found for #__extensions table

Expectation failed for method name is equal to <string:parseSchemaUpdates> when invoked 1 time(s).
Method was expected to be called 1 times, actually called 0 times.
```

* cs-fix extra semicolon

* remove trailing coma

* remove trailing comma

marcochirienti committed:
Coding style - PHP code tag and control structures II (#13149)

* Removed unnecessary spaces after PHP code opening tag

* Removed unnecessary spaces before PHP code closing tag

* Added a white-space before PHP code closing tag where missing

* Fixed style for control structures alternative syntax

zero-24 committed:
Remove never needed nor used JED Update Server from new installs (#13047)

* Update joomla.sql

* Update joomla.sql

* Update joomla.sql

* Update joomla.sql

* Update joomla.xml

* Update jos_update_sites.csv

* Update jos_update_sites_extensions.csv

* Update script.php

* Update script.php

* Update joomla.sql

* Update script.php

andrepereiradasilva committed:
[PHP 7.1] Adapt JDate to reflect that php now sets DateTime() with microseconds (#13079)

* Update date.php

* Update date.php

* Update date.php

frankmayer committed:
Replacements for deprecated JFile::read() (#13160)

* Replacements for deprecated JFile::read()

* Removed replacements for FOF

frankmayer committed:
Type-safe comparison for isClient (#13157)

AlexRed committed:
Set correct default values for user creation in the backend also in the sample_learn.sql sample data (#13127)

The default value for the useractivation param is 2
https://github.com/joomla/joomla-cms/blob/3.6.4/administrator/components/com_users/config.xml#L45-L68

AlexRed committed:
Set correct default values for user creation in the backend also in the sample_learn.sql sample data (#13127)

The default value for the useractivation param is 2
https://github.com/joomla/joomla-cms/blob/3.6.4/administrator/components/com_users/config.xml#L45-L68

frankmayer committed:
Removed some unnecessary strtotime() calls (#13171)

csthomas committed:
Fix unittest on cache - 5 seconds less (#11872)

* Run test at JCacheTest->testGc with not empty storage and caching enabled.

* Test if cache file after run gc does not exists.

frankmayer committed:
Fixed misuses of array_push() project wide (#12170)

* Replaced rand() with mt_rand() projectwide

* Fixed misuses of array_push()

* Fixed PHPMailer problems

* Reverted some changes, as they are in included external frameworks

* Removed changes which were irrelevant to this PR.

* Changed some recently added array_push calls

* Fixed accidental deletion of curly bracket

* revert e91177da1316d53da3fcc9878c5220bbbfd0bad2

chrisdavenport committed:
Smart search parser leaking CSS and JS terms (#12411)

* Strip head first because of possible nested tags

* Remove chunking from indexer

ggppdk committed:
Responsive site title for protostar (#13064)

* Responsive site title for protostar

* Updated less file too

* Fixed less compling, added maximum of 40px for font-size

* Better scaling factor, removed calc on line-height it is not really needed and causes a little jumping

bembelimen committed:
Improve menu associations (#13086)

csthomas committed:
Unittest fix - do not use PDO driver directly (#13097)

* Use database driver in the right way in unittests

* Fix unittests

Commit:
Allow packages to declare that their child extensions cannot be uninstalled (#13154)

* Allow packages to declare that their child extensions cannot be uninstalled

* Change tag name, fix bugged instanatiation of manifest class, add tag to English package manifest

* Remove language check

* Flag in JInstaller that a package is being uninstalled to allow its children to be removed

csthomas committed:
Speed up hhvm unit tests - disable jit (#13169)

arrowthemes committed:
Wrong color on active menu item in mobile view - Protostar (#12303)

* Update template.less

Fix css issue with menu active item in mobile screens

* Update template.css

CSS Fix for active menu item in mobile screens

laoneo committed:
[com_fields] Include group state only when not on back end list (#12696)

* Include group state only when not on back end list

* Add comment

zero-24 committed:
Update default_permissions.php

zero-24 committed:
Merge branch 'staging' into fix_13083

frankmayer committed:
Remove senseless proxy functions

Commit:
Updating calendar locales

frankmayer committed:
Replacements for deprecated getCfg() (#13161)

* Replacements for deprecated getCfg()

* Replacements for deprecated AbstractRegistryFormat Class implementations

bembelimen committed:
Don't load authorised view levels in foreach for tags generation (#13140)

* Don't load authorised groups in foreach for tags generation

* Add $

* Use user method

* Change variable name

frankmayer committed:
Replacements for deprecated JApplication::getHash() (#13162)

* Replacements for deprecated JApplication::getHash()

* Replacements for deprecated JApplication::getHash()

* Replacements for deprecated JApplication::getHash()

marcochirienti committed:
Coding style - PHP code tag and control structures III (#13173)

* Removed unnecessary spaces after PHP code opening tag

* Added a white-space before PHP code closing tag where missing

* Removed unnecessary spaces before PHP code closing tag

* Fixed style for control structures alternative syntax

* Fixed style for control structures alternative syntax

* Added a semicolon at the end of statement where missing

marcochirienti committed:
Coding style - PHP code tag and control structures IV (#13176)

* Removed unnecessary spaces after PHP code opening tag

* Removed unnecessary spaces before PHP code closing tag

* Added a white-space before PHP code closing tag where missing

* Removed unnecessary spaces before PHP code closing tag

* Added a white-space before PHP code closing tag where missing

* Added a white-space before PHP code closing tag where missing

* Added a semicolon at the end of statement where missing

* Fixed style for control structures alternative syntax

laoneo committed:
Copy and new should use the group_id (#13179)

frankmayer committed:
Unnecessary double quotes in site components (#13183)

* Replace unnecessary double quotes in site/components

* small correction

* Included a fix for an existing bug that @andrepereiradasilva pointed out

frankmayer committed:
Merged unset calls (#13186)

frankmayer committed:
Remove one-time-use variables (#13187)

frankmayer committed:
Remove unnecessary parentheses in site components (#13190)

* Remove unnecessary parentheses in site/components

* Insert space, wrongfully removed

* Re-added a pair, for readability as @andrepereiradasilva suggested not to have it ungrouped for readability

* Removed some whitespace, as noted in @shur's comment

frankmayer committed:
Use short syntax for applied operations (#13188)

infograf768 committed:
New Feature: add redirect option after article submission

laoneo committed:
Remove DPFields leftover (#13229)

laoneo committed:
[com_fields] Migrate from repeatable to subform field for list options (#13069)

* Migrate from repeatable to subform field for list options

* Adapt checkboxes and radio

* Merge regression

frankmayer committed:
Simplify ternary with elvis in site component (#13185)

* Simplify ternary with elvis operator

* Fixed indentation

* Forgot to remove these parentheses in merge from staging

frankmayer committed:
Removed silly assignments (#13196)

frankmayer committed:
Replace unnecessary double quotes (#13239)

frankmayer committed:
Replace unnecessary double quotes in libraries/cms (#13235)

* Replace unnecessary double quotes

* Reverted some double quotes-replacements

frankmayer committed:
Various changes in modules (#13216)

* Simplify ternary operation with elvis operator

* Remove unnecessary parentheses

* Remove redundant variables

* Code formatting

coolcat-creations committed:
Add cursor:pointer; for making dismiss alert work on ios devices (#11965)

frankmayer committed:
Simplify ternary operation using elvis operator

frankmayer committed:
Remove unneeded parentheses

ciar4n committed:
Media Manager Restyle (#12643)

* Media manager restyle - Initial commit

* Media Manager restyle

* Delete option added on image hover

* Center thumbnail X icon

* Media restyle added to xtd-button

* JS moved popup-imagemanager.js

* JS added to popup-imagemanager.min.js

* CS

* Revert iframe height increase

* Modal margin

* RTL support

* RTL - correcting image popup display

zero-24 committed:
Merge branch 'staging' into fix_13083

mbabker committed:
Merge tag '3.6.5' into staging

andrepereiradasilva committed:
cs fix travis built (#13255)

infograf768 committed:
Deleting unused lang strings (com_fields) (#13259)

* Deleting unused language string

* another

zero-24 committed:
Merge branch 'staging' into fix_13083

frankmayer committed:
Cleanups, fixes and a bit of optimizations for site/components batch #3 (#12292)

* Cleanups, fixes and a bit of optimizations for site/components batch #3

- com_content

Note: This is a single commit bundling all types of changes, since PR #12261 which had detailed commits, was rejected as a whole

* Ch-Ch-Ch-Changes!

Made some changes as pointed out by @andrepereiradasilva

* A bit more...

* Revert

* Removed empty function, as there is a fallback. Change made according to comment from @wilsonge

* Included @andrepereiradasilva's suggestions

* Inserting whitespace before php closing tag

frankmayer committed:
Replace unnecessary double quotes in templates (#13245)

* Replace unnecessary double quotes

* Revert overzealous replacements

* Formatting fix

frankmayer committed:
Replace unnecessary double quotes (#13215)

frankmayer committed:
Cleanups, fixes and a bit of optimizations for site/components batch #1 (#12290)

* Cleanups, fixes and a bit of optimizations for site/components batch #1

- com_ajax
- com_banners
- com_config
Note: This is a single commit bundling all types of changes, since PR #12261 which had detailed commits, was rejected as a whole

* Ch-Ch-Changes!

* Removed the query changes

* Fix for wrong merge resolve

frankmayer committed:
Replaced unnecessary double quotes in plugins (#13211)

* Replaced unnecessary double quotes in plugins

* spaces

frankmayer committed:
Revert change to 3rd party file.

frankmayer committed:
Removed unnecessary parentheses in plugins (#13212)

frankmayer committed:
Various changes in plugins (#13213)

* Remove one-time use variable

* Merge unset calls

* Remove obvious variants from ternary operation

* Use short syntax for applied operation

wilsonge committed:
Merge pull request #13244 from frankmayer/elvis-and-unneeded-parentheses-in-templates

Simplify some ternary operations using elvis operator and remove unnecessary parentheses in templates

wilsonge committed:
Merge pull request #13193 from zero-24/fix_13083

[Fix #13083]Text Filters text shown on Permissions tab in Global Configuration

wilsonge committed:
Merge pull request #12802 from AlexRed/patch-19

delete old files from Joomla 3.2.3 version

wilsonge committed:
Merge pull request #13228 from infograf768/redirectaftersubmit

New Feature: add redirect option after article submission

wilsonge committed:
Merge pull request #13198 from frankmayer/remove-senseless-proxy-functions

Remove senseless proxy functions

wilsonge committed:
Merge pull request #12642 from alikon/patch-77

[com_fields] SQL - FIND_IN_SET() portable other than mysql

frankmayer committed:
Merge branch 'staging' into issite-conversion

# Conflicts:
#	components/com_config/controller/helper.php
#	plugins/system/debug/debug.php

frankmayer committed:
Performance 5 (modules) (#12229)

* Use Elvis instead of full ternary

* Included nested positive if in parent

* Double quotes to single quotes

* Removed unnecessary parentheses

* Inline one-time use variables

* Type-safe string comparisons

* More type safe comparisons

* Replace is_null() with === null

* Replace strstr with strpos

* Replace substr with strpos

* Optimize non-optimal if conditions.

* Merged cascading str_replace calls

* Another type safe-comparison

* Fix multiple calls to callable in if condition

* Flip some conditions back after removing the condition call

* Fix codesniffer

* Fix codesniffer...

* This one is probably better left loose typed

* ... and this one, too.

* Use Elvis instead of full ternary

* Included nested positive if in parent

* Double quotes to single quotes

* Removed unnecessary parentheses

* Inline one-time use variables

* Type-safe string comparisons

* More type safe comparisons

* Replace is_null() with === null

* Replace strstr with strpos

* Replace substr with strpos

* Optimize non-optimal if conditions.

* Merged cascading str_replace calls

* Another type safe-comparison

* Fix multiple calls to callable in if condition

* Flip some conditions back after removing the condition call

* Fix codesniffer

* Fix codesniffer...

* This one is probably better left loose typed

* ... and this one, too.

* A few more changes based on discussion with @andrepereiradasilva

andrepereiradasilva committed:
Drop index idx_image in #__languages table (#13252)

* Update joomla.sql

* Update joomla.sql

* Update joomla.sql

* Update 3.7.0-2016-11-21.sql

* Update 3.7.0-2016-11-21.sql

* Update 3.7.0-2016-11-21.sql

Commit:
HTML5 pattern in validate.js (#13094)

* html5 pattern

* implement @ralain 's suggestion

Thank you @ralain

frankmayer committed:
Fixes for errors introduced with conflict resolution

frankmayer committed:
Fixes for errors introduced with conflict resolution

andrepereiradasilva committed:
Update head.php (#13081)

laoneo committed:
[com_fields] Change extension variable to context for custom field groups (#13175)

* Change extension variable to context for custom field groups

* Stabilizing javascript

* Fixing code style error

* Check for admin permission of com_fields access correct

Commit:
Don't process value adjusting twice (#13233)

ciar4n committed:
Modals style cleanup (#13261)

frankmayer committed:
Variable names - naming conventions - testing the waters (#13170)

frankmayer committed:
Various changes in libraries legacy (#13241)

* Inline one-time use variables

* Removed unnecessary ternary operators

* Merge unset() calls

* Use modern day type-casting

* This can be replaced with type-casting

frankmayer committed:
Changes on continue statements (#13199)

* Remove unnecessary 'continue' function

* Change 'continue' so that it breaks out of the 'switch', as there is no need to compare against the rest.

* Fixed mistake

* Added a break in `separator` case, as there is nothing to execute under `heading`,  in that case.

* Revert changes in 3rd party file

ciar4n committed:
Fix vertical scroll (#13269)

frankmayer committed:
Type safe string comparison of strings in plugins

mbabker committed:
Remove false param on fetching form data (#13254)

Commit:
Remove unneeded registry creation (#12571)

* Contact params already are Registry objects.

* Newsfeed params already are Registry objects.

* Menu params already are Registry objects.

* Fixing wrong merging

mbabker committed:
Lazy session starting (#13074)

* Allow lazy starting a session

* Defer starting sessions in the installer

* Defer starting session in JApplicationCms

* Unit test fixes

wilsonge committed:
Merge pull request #13271 from frankmayer/type-safety-in-plugins

Type safe string comparison of strings in plugins

Commit:
[com_fields] Normalise data structure for assigned categories (#13246)

* Normalise #__fields -> assigned_cat_ids

* Fixing typo in table name

frankmayer committed:
Merge branch 'staging' into issite-conversion

# Conflicts:
#	plugins/system/languagefilter/languagefilter.php
#	plugins/user/profile/field/dob.php
#	plugins/user/profile/profile.php

pocketpc committed:
Fixed issue #11932

Removed an unnecessary range restriction on a SQL query, which caused an
invalid query to be generated for Microsoft SQL Server.

zero-24 committed:
fix CS after 3.6.5 ..

frankmayer committed:
Fix accidentally flipped comparison

frankmayer committed:
Replace unnecessary double quotes

wilsonge committed:
Merge pull request #13156 from frankmayer/issite-conversion

Replace deprecated isSite() and isAdmin() calls

frankmayer committed:
some more double quote replacements

frankmayer committed:
This one, too...

wilsonge committed:
Merge pull request #13290 from zero-24/beez_cs

[CS] Fixes CS after the 3.6.5 changes in the beez override

wilsonge committed:
Merge pull request #13292 from frankmayer/unnecessary-double-quotes-in-administrator-no-1

Replace unnecessary double quotes in administrator #1

Commit:
Add context selector to fields groups

wilsonge committed:
Merge pull request #13288 from pocketpc/staging

Fixed issue #11932 (Mssql: article can´t be created)

laoneo committed:
Composer install

wilsonge committed:
Merge pull request #13303 from Digital-Peak/restore-vendor

[com_fields] Vendor directory restore

frankmayer committed:
Type-safe string comparison in site/templates

rdeutz committed:
prepare alpha1 release

laoneo committed:
[com_fields] Migrate user groups from categories to field groups (#13103)

* Migrate user groups from categories to field groups

* Renamed variable from category to group

* Show only fields which are in selected groups

* Change identifier from extension to context

frankmayer committed:
Merge branch 'staging' into type-safe-string-comparisons-in-site-templates

# Conflicts:
#	templates/beez3/html/com_contact/contact/default_user_custom_fields.php

wilsonge committed:
Merge pull request #13301 from Bakual/AddingContextSelectToFieldsGroups

[com_fields] Add context selector to fields groups

wilsonge committed:
Merge pull request #13309 from frankmayer/type-safe-string-comparisons-in-site-templates

Type-safe string comparison in site/templates

alikon committed:
[com_fields] - Creating a New field give SQL error on postgresql (#13155)

* [com_fields] - Creating a New field give SQL error on postgresql

Postgres: Inserting an image field fails with an error #13137

* xml default

xml default

* added JNONE

added JNONE

ciar4n committed:
Template menu assign fix n styling (#13280)

mbabker committed:
Improve doc blocks for JMenu (#13274)

frankmayer committed:
Moved JPluginHelper::importPlugin() call outside loops. (#13248)

mbabker committed:
Logout code needs to account for shared_session parameter (#13273)

okonomiyaki3000 committed:
Fix subform chosen (#12993)

* Initialize chosen when adding rows in repeatable subforms

* Use JHtml::script and JHtml::stylesheet in the new way

* Subform does not need to fix chosen.

ciar4n committed:
Codemirror fluid height (#13266)

* Codemirror fluid height

* Min-height changed to 400px

* Moved less to _editors.less - added max-height

Commit:
Delete assigned categories when field is deleted (#13315)

frankmayer committed:
Remove unused local variables project wide (#13206)

* Remove unused or immediately overwritten variables

* Remove unused or immediately overwritten variables in tests

* Commented out some reference code, as it is not used in that scope

* Not sure on this removal. Added a TODO.

* Fixed accidentally removed $

* Fix codesniffer

* Revert changes in this file, as it is an external library

* Update field.php

Commit:
Fix xtd-buttons (#13323)

Commit:
[Easy fix] Typo - a trademark should not be altered

Easy fix on review.
As a trademark, `The Joomla! Project`, even in core language strings should not be altered (no exceptions as a legal matter).
I know no one cares (and TBH me neither) but you can't ask others to respect a rule if we don't.

Previous:
`The Joomla project`
New:
`The Joomla! Project`

jeckodevelopment committed:
Merge pull request #13329 from MATsxm/patch-3

[Easy fix] Typo - a trademark should not be altered

infograf768 committed:
[com_fields] Normalising title strings (#13328)

infograf768 committed:
Correcting bug introduced in #12932 (#13326)

cpfeifer committed:
[com_fields] Field Edit UI Adjustment (#13333)

* Field Edit UI Adjustment

Place options tab 2nd instead of 3rd

* Updating code to correct 3rd and 4th tab positions

Updating code to correct 3rd and 4th tab positions

zero-24 committed:
alpha2 development (#13347)

Commit:
Remove assigned_cat_ids from FieldsTable (#13334)

laoneo committed:
Render only the input when no fieldsets are specified in subform layouts (#13093)

* Render only the input when no fieldsets are specified in subform layouts

* Reset margin left, because we don't have a label

* Better selector

* Move CSS margin reset to the main layout

* Read missing new line

mbabker committed:
Quick review on JLog (#13360)

Fedik committed:
TinyMCE tolbar builder: use TinyMCE translation for the TinyMCE buttons (#13368)

* Use TinyMCE translation for the TinyMCE buttons

* One more unused string

Fedik committed:
fix TinyMCE automatic language detection (#13366)

laoneo committed:
[com_fields] Move the after title event to the correct position (#13345)

* Move the after title to the correct position

* Move the fields outside of the page header

frankmayer committed:
Few changes in installation (#13297)

* Unnecessary double quotes

* dirname to __DIR__

* Optimized loop termination condition

* Short syntax for applied operation

* Remove one-time-use variable

* Remove superfluous property initialization

* DO NOT REPLACE dirname() with __DIR__ in installation because of B/C with 5.2!!!
DO NOT REPLACE dirname() with __DIR__ in installation because of B/C with 5.2!!!
DO NOT REPLACE dirname() with __DIR__ in installation because of B/C with 5.2!!!
DO NOT REPLACE dirname() with __DIR__ in installation because of B/C with 5.2!!!

frankmayer committed:
Replace unnecessary double quotes (#13293)

frankmayer committed:
Replace unnecessary double quotes in administrator #3 (#13295)

* Replace unnecessary double quotes

* And one unnecessary semicolon... ;)

* CS Fix + removal of unnecessary space

* some more double quote replacements

* Changes, based on @shur's comments

frankmayer committed:
Remove unnecessary parentheses in installation (#13298)

mbabker committed:
Update Composer libraries to current versions (#13321)

andrepereiradasilva committed:
add missing echo (#13374)

davidjosephhayes committed:
Add more flexibility to JTable ordering (#12464)

* Set JTable and JModelAdmin to use the _columnAlias propery of JTable for ordering

* fix concat spacing

* update code comments

* add quoted ordering field as var to prevent calling quoteField multiple times; fix missing quoteField

* add jtable ordering abstraction layer to jtablenested

* fix incorrect negation

* Update nested.php

* Update table.php

* Update admin.php

cpfeifer committed:
[com_fields] Update Custom Field option label (#13378)

* Merge remote-tracking branch 'joomla/staging' into staging

* Update Custom Field Option Parameter Label

andrepereiradasilva committed:
remove extra space (#13381)

Commit:
[com_fields] Remove GroupConcat from fields model (#13302)

* Remove GroupConcat

* Adding FieldsHelper::getAssignedCategoriesTitles

* Codestyle

Commit:
Merge FieldsHelperInternal into FieldsHelper (#13316)

Commit:
[com_fields] Fixing "Search - Content" plugin and make it work with fields (#13348)

* Fixing "Search - Content" plugin and make it work with fields

* Make sure types match for PostgreSQL

andrepereiradasilva committed:
installation update also modified by field and fields table (#12947)

ggppdk committed:
Fix custom fields gallery field in windows installations (#12844)

frankmayer committed:
Further work on layouts (#13296)

* Replace isArray()... with type-cast

* Type-casting instead of strVal()

* Use elvis!

* Ternary operator simplification and type safe comparison

* Unnecessary parentheses

chrisdavenport committed:
Load system plugins in Smart Search indexer (#13358)

Commit:
Updating languages for calendar

mbabker committed:
Update to PHPMailer 5.2.21 (#13388)

andrepereiradasilva committed:
Execute sql uninstall query (if exists) on extension install abort (#13376)

* add sql rollback

* execute uninstall query on install abort

* removed unused test

frankmayer committed:
Some improvements in tests #1:
- removed senseless statements
- one case correction
- one loop optimization
- removed a closing tag

frankmayer committed:
Remove identical method override

frankmayer committed:
Some improvements in tests #4:
- removal of some unused variables (and inserted todo's if unclear)
- removal of null assignments on class properties;

frankmayer committed:
Some improvements in tests #2: (#13398)

* Some improvements in tests #2:
- fix callable name mismatches
- removed closing tags (also one that's in other PR, to not clash)
- removed loop that does nothing

* Revert removal of empty loop, as $i is decremented in the while statement

frankmayer committed:
2 to 1 loop optimization in plugins/search/categories/categories.php (#13390)

* Another lonely `for` to `foreach`modification...

* Too much code for today... not seing the obvious... :smile:

* Implementing a good catch by @andrepereiradasilva

* Optimized loop further

wilsonge committed:
Merge pull request #13391 from frankmayer/improvements-in-tests-1

Some improvements in tests #1:

wilsonge committed:
Codestyle

wilsonge committed:
Fix cast statements needing a space

wilsonge committed:
Fix concat operator spaces

jeckodevelopment committed:
Merge pull request #13411 from wilsonge/ut-cast-statements

[Unit Test CS] Cast Statements and Concat Operator spacing issues

wilsonge committed:
Spaces instead of tabs issues (#13409)

Commit:
Updating languages for calendar

ciar4n committed:
Toolbar min-height correction (#13413)

ralain committed:
Fix double it typo (#13418)

ciar4n committed:
Add thumbsnails-media class (#13406)

Fedik committed:
TinyMCE builder: use usergroup instead of viewlevel for each set. Plus fix some BC issue. (#13387)

* Use user groups instead of view level for TinyMCE builder

* Preload an old parameters where it possible, for BC

* cs

* onDisplayLegacy for BC

* space to tabs

* Warning message about update

* Show warning only in the backend and only twice per session

* Allow to select the user group only once per the set

* Too much code :alien:

* Update the warning message

* Update the warning message

* Update tinymce.php

Correcting B/C

Harmageddon committed:
Use JPATH_ADMINISTRATOR instead of JPATH_ROOT/administrator.

frankmayer committed:
Fix,  according to @mbabker's comment

frankmayer committed:
Another fix,  according to @mbabker's comment

wilsonge committed:
Merge pull request #13422 from Harmageddon/jpath_administrator

Use JPATH_ADMINISTRATOR instead of JPATH_ROOT/administrator.

wilsonge committed:
Merge pull request #13408 from frankmayer/improvements-in-tests-4

Some improvements in tests #4:

photodude commented:
@piotr-cz can you fix the merge conflicts please... once that is done this PR can move forward

andrepereiradasilva committed:
[Deprecate _QQ_ 1] Remove information about _QQ_ in 4 ini files. (#13426)

* Update en-GB.ini

* Update en-GB.lib_joomla.ini

* Update en-GB.ini

* Update en-GB.lib_joomla.ini

photodude committed:
code style fix for administrator/components/com_contenthistory (#13439)

Automatically fixed with PHPCS2 fixers

andrepereiradasilva committed:
remove deprecated calls (#13424)

ciar4n committed:
Container added to all batch modals (#13307)

* Container added to all batch modals

* Fix overflow & com_modules

mbabker committed:
Do not include the SQL query in the Exception message (#13356)

cpfeifer committed:
[com_fields] Cf permission strings (#13384)

* Merge remote-tracking branch 'joomla/staging' into staging

* com_fields language string update

* com_fields language string update

* com_fields language string update

* Small text edit

chrisdavenport committed:
Move weblinks stats to weblinks repo (#13389)

andrepereiradasilva committed:
[JLanguage] Simplify language file parse (#13407)

* simplify language file parse

* revert $php_errormsg

* as suggested by michael

alikon committed:
[postgresql] - no need to set default to NULL (#13419)

Fedik committed:
Make sure that the Public group used only once (#13434)

infograf768 committed:
Correcting default TinyMCE params when installing Joomla (#13431)

Fedik committed:
Add some description to TinyMCE builder, to make it more clear. (#13367)

* Add the description for the source and target sections in TinyMCE builder.

* Spaces to Tabs

* normalise "drag and drop"

* caps for label

* Improve the label for an access field in the set

* Update strings

piotr-cz committed:
Merge branch 'convert-crlf' of https://github.com/piotr-cz/joomla-cms into convert-crlf

# Conflicts:
#	installation/language/fa-IR/fa-IR.ini

piotr-cz commented:
Rebased on latest staging branch again.

laoneo commented:
You still want to merge 890 commits with over 2000 file changes? Did you rewrite Joomla during holiday :wink: ?

infograf768 commented:
This PR also changes some stuff else than EOL

rdeutz commented:
This PR does more than it says in the desc. It needs to be re-done

piotr-cz commented:
I'll prepare new one then
